services/web-server/src/PulseEngine/PulseIterator.js

```diff
@@ -14,22 +14,25 @@ export default class PulseIterator {
     this.listening = true;
     this.subscriptionId = this.pulseEngine.subscribe(
       subscriptions,
-      this.pushValue.bind(this)
+      this.pushValue.bind(this),
+      this.pushError.bind(this),
     );
+    this.pullId = 0;
   }
 
   next() {
-    return this.listening ? this.pullValue() : this.return();
+    return this.pullValue();
   }
 
   return() {
-    this.emptyQueue();
+    this.stop({done: true});
 
-    return Promise.resolve({ value: undefined, done: true });
+    // return should always succeed, even if an error was introduced
+    return Promise.resolve({done: true});
```

Comment:
can you tell me more about the change `{ value: undefined, done: true }` ==> `{done: true}`?

Reply:
(I think on the other end a "tuple" is expected, hence my question)

Reply:
Just saving a bit of code, really - the two should be equivalent.  I think I changed this a few times in hacking on it.  The [iterator protocol docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols) say that the value can be omitted if `done` is `true`.